packages/ragbits-core/src/ragbits/core/audit/cli_tracer.py

```diff
@@ -0,0 +1,65 @@
+from rich.console import Console
+
+from ragbits.core.audit import TraceHandler
+from ragbits.core.audit.cli_span import CLISpan
+
+
+class CLITracer(TraceHandler[CLISpan]):
+    """
+    CLITracer class for all trace handlers.
+    """
+
+    def __init__(self) -> None:
+        """
+        Construct a new CLITracer instance.
+        """
+        super().__init__()
+
+    def start(self, name: str, inputs: dict, current_span: CLISpan | None = None) -> CLISpan:
+        """
+        Log input data at the beginning of the trace.
+
+        Args:
+            name: The name of the trace.
+            inputs: The input data.
+            current_span: The current trace span.
+
+        Returns:
+            The updated current trace span.
+        """
+        parent = self._spans.get()[-1] if self._spans.get() else None
```

Comment:
no need to use private stack here, we can use `current_span` argument it should contain the same node

Reply:
it does. Thanks. Corrected